File: SherlockAndSquares.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */   
    long long int T;
    cin>>T;
    long long int A,B; 
    while(T--)
        {
        cin>>A>>B;
        long long int A_min = floor(sqrt(A));
        long long int count = 0;
        long long int A_min_square = A_min * A_min;
        do
            {
            
            if(A_min_square >= A && A_min_square <= B)
                count++;
            A_min++;
            A_min_square = A_min * A_min;
        }while(A_min_square<=B);
            cout<<count<<endl;
    }
    return 0;
}

File: SherlockAndSquares.go
package main
import (
    "fmt"
    "math"
)

func main() {
 //Enter your code here. Read input from STDIN. Print output to STDOUT
    var t int
    var sum int
    fmt.Scanf("%v",&t)
    for ;t>0;{
        sum = 0
        var x,y int
        fmt.Scanf("%v%v",&x,&y)
        var i = int(math.Sqrt(float64(x)))
        for ;i*i<=y;i++{
            if i*i >= x && i*i <= y{
                sum += 1
            }
        }
        t--
        fmt.Println(sum)
    }
}
